judoscale/flask/judoscale.py

```diff
@@ -0,0 +1,26 @@
+from flask import request
+
+from judoscale.core.config import config as judoconfig
+from judoscale.core.metric import RequestMetrics
+from judoscale.core.metrics_store import metrics_store
+from judoscale.core.reporter import reporter
+
+
+def report_metrics():
+    request_start_header = request.headers.get("x-request-start", "")
+    request_metric = RequestMetrics(request_start_header)
+    metric = request_metric.get_queue_time_metric_from_header()
+    if metric:
+        metrics_store.add(metric)
+    reporter.ensure_running()
+    return None
+
+
+class Judoscale:
+    def __init__(self, app):
+        if app is not None:
+            self.init_app(app)
+
+    def init_app(self, app):
+        judoconfig.merge(app.config.get("JUDOSCALE", {}))
```

Comment:
I think we might _also_ want to call `reporter.ensure_running()` here. We do [something similar](https://github.com/judoscale/judoscale-python/blob/bd0cdf3ef77137b539f1bc4497b97cc19bdc5e92/judoscale/django/apps.py#L23) in our Django integration to ensure the reporter is started as early as possible.

This is helpful for apps that get very little traffic, such as staging or demo apps. Customers often use these apps to experiment with Judoscale, and it's confusing for them if they install the adapter, but the adapter isn't recognized by the Judoscale service. This can happen if the app hasn't received traffic and the reporter hasn't started yet. By starting the reporter when the app boots, we ensure the service "sees" the adapter as early as possible.

Note that we'll still need to call `reporter.ensure_running()` when handling the request, because we need to handle forking web servers. I'm actually not sure if Flask can be run that way, but my understanding is that Gunicorn is a forking server similar to Puma in Ruby, and so a reporter started when the app boots will be running in a different process than the process that serves requests.

Does this all make sense?

Reply:
Makes perfect sense, yep. I've updated `.init_app()` to also ensure the reporter is running.

Gunicorn is a [forking web server](https://docs.gunicorn.org/en/stable/design.html#server-model), which — like you said — basically means that there will be a master process managing a handful of worker processes which are handling HTTP requests. For instance if there are 2 worker processes, we'd have 3 reporters running — one for the master process and one for each worker process. The master process will never execute the `report_metrics()` callback because it's not handling HTTP requests, so it'll continuously report empty metrics.